static/app/components/modals/savedSearchModal/editSavedSearchModal.spec.tsx

```diff
@@ -0,0 +1,144 @@
+import selectEvent from 'react-select-event';
+
+import {render, screen, userEvent, waitFor} from 'sentry-test/reactTestingLibrary';
+
+import {
+  makeClosableHeader,
+  makeCloseButton,
+  ModalBody,
+  ModalFooter,
+} from 'sentry/components/globalModal/components';
+import {EditSavedSearchModal} from 'sentry/components/modals/savedSearchModal/editSavedSearchModal';
+import {SavedSearchType, SavedSearchVisibility} from 'sentry/types';
+import {IssueSortOptions} from 'sentry/views/issueList/utils';
+
+describe('EditSavedSearchModal', function () {
+  beforeEach(function () {
+    MockApiClient.clearMockResponses();
+
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/recent-searches/',
+      method: 'GET',
+      body: [],
+    });
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/recent-searches/',
+      method: 'POST',
+      body: [],
+    });
+  });
+
+  const defaultProps = {
+    Body: ModalBody,
+    Header: makeClosableHeader(jest.fn()),
+    Footer: ModalFooter,
+    CloseButton: makeCloseButton(jest.fn()),
+    closeModal: jest.fn(),
+    organization: TestStubs.Organization({features: ['issue-list-saved-searches-v2']}),
+    savedSearch: {
+      id: 'saved-search-id',
+      name: 'Saved search name',
+      query: 'is:unresolved browser:firefox',
+      sort: IssueSortOptions.DATE,
+      visibility: SavedSearchVisibility.Owner,
+      dateCreated: '',
+      isPinned: false,
+      isGlobal: false,
+      type: SavedSearchType.ISSUE,
+    },
+  };
+
+  it('can edit a saved search with org:write', async function () {
+    const editMock = MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/searches/saved-search-id/',
+      method: 'PUT',
+      body: {
+        id: 'saved-search-id',
+        name: 'test',
+        query: 'is:unresolved browser:firefox',
+        sort: IssueSortOptions.PRIORITY,
+        visibility: SavedSearchVisibility.Owner,
+      },
+    });
+
+    render(<EditSavedSearchModal {...defaultProps} />);
+
+    userEvent.clear(screen.getByRole('textbox', {name: /name/i}));
+    userEvent.type(screen.getByRole('textbox', {name: /name/i}), 'new search name');
+
+    userEvent.clear(screen.getByTestId('smart-search-input'));
```

Comment:
We're probably not labeling the input right if you can't use the field label.

You need to pass down the `id` to the component I think

Reply:
Ah so I think there is something slightly off about the aria-label in the FormField component. Because RTL is telling me that there are two different labels with the same text (this is after passing the correct ID in). But removing the aria-label causes it to work as intended. I'll look into this a bit more once saved searches is done